### ISSUE TITLE
The parser breaks when using html template tag

### ISSUE THREAD

**Issue by abdonrd** [stylelint/stylelint#4800]
> Clearly describe the bug

The parser breaks when using html template tag

> Which rule, if any, is the bug related to?

None.

> What code is needed to reproduce the bug?

```javascript
export class X extends Y {
  static styles = css`
    :host {
      display: block;
      display: block;
    }
  `;

  render() {
    return html`<div>Hi!</div>`;
  }
}

```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-duplicate-properties": [
      true,
      {
        "ignore": [
          "consecutive-duplicates-with-different-values"
        ]
      }
    ]
  }
}
```

> Which version of stylelint are you using?

`v13.5.0`.

It works well with the `v13.3.3`. Stop working at `v13.4.0`.

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.ts"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Template literals.

> What did you expect to happen?

No parser error

> What actually happened (e.g. what warnings or errors did you get)?

The parser breaks when using html template tag:

```shell
> stylelint "client/src/pages/**/*.ts"

TypeError: result.apply is not a function
    at /Users/abdonrd/pwa-lit-template/node_modules/@stylelint/postcss-css-in-js/extract.js:274:20
    at Array.some (<anonymous>)
    at isStylePath (/Users/abdonrd/pwa-lit-template/node_modules/@stylelint/postcss-css-in-js/extract.js:260:33)
    at TaggedTemplateExpression (/Users/abdonrd/pwa-lit-template/node_modules/@stylelint/postcss-css-in-js/extract.js:359:8)
    at NodePath._call (/Users/abdonrd/pwa-lit-template/node_modules/@babel/traverse/lib/path/context.js:55:20)
    at NodePath.call (/Users/abdonrd/pwa-lit-template/node_modules/@babel/traverse/lib/path/context.js:42:17)
    at NodePath.visit (/Users/abdonrd/pwa-lit-template/node_modules/@babel/traverse/lib/path/context.js:90:31)
    at TraversalContext.visitQueue (/Users/abdonrd/pwa-lit-template/node_modules/@babel/traverse/lib/context.js:112:16)
    at TraversalContext.visitSingle (/Users/abdonrd/pwa-lit-template/node_modules/@babel/traverse/lib/context.js:84:19)
    at TraversalContext.visit (/Users/abdonrd/pwa-lit-template/node_modules/@babel/traverse/lib/context.js:140:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pwa-lit-template@0.1.0 lint:stylelint: `stylelint "client/src/pages/**/*.ts"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pwa-lit-template@0.1.0 lint:stylelint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/abdonrd/.npm/_logs/2020-05-22T11_59_28_821Z-debug.log
```

**Comment by abdonrd** [comment]
After check the diff from https://github.com/stylelint/stylelint/compare/13.3.3...13.4.0:

The error seems related to this:

https://github.com/stylelint/stylelint/compare/13.3.3...13.4.0#diff-86e39c31ff3c9300102b40c22b3d8c4a

Because before this release, the `html` syntax was `'postcss-html'`, then the `result` was `false` here:

https://github.com/stylelint/postcss-css-in-js/blob/7a28eabd8851114b3394b89c8d510bd1da2e1690/extract.js#L256-L274

And now enter here and breaks:

https://github.com/stylelint/postcss-css-in-js/blob/7a28eabd8851114b3394b89c8d510bd1da2e1690/extract.js#L265

**Comment by abdonrd** [comment]
Here you a demo:

**Note: The demo is working but probably because the playground is not using the last Stylelint version?**

https://stylelint.io/demo#N4Igxg9gJgpiBcIYA8AOEBOAXABGANgIYDOxOAGjiljAHZRkCaOwAOrTjsVoVgJZguWAJ74YZALx5SAA3adO8ABYRuLeQs5Q+xVEWHwcAI3wQwAawDcGzdt37DJs1Zs4AvhpnXaGjHVgYABQAlOocCn5YAK4YHEpYALb4MgA82gBuAHwAEnwAhCkA9BmZXhoetBUgADQgxMK0PMgI4KQAtHy0bQBWxDXgELQAZnwA5i1s4awgGFFixNOGk5rTsASEGLx8g21OFm20EG1QUXoCvDBtqBgQqDDYfOKLOADarpxYszDV72GaKyAxoc-M83uF-gpppBaMQYGAovx0pcTmcwBdiG0AO58LBKY58IZDe50LBtdKEfBRJ4gX6cAC6vwqmgZ4QqbhAbiAA

**Comment by jeddy3** [comment]
@abdonrd Thank you for digging into this. @m-allanson continued [to investigate](https://github.com/stylelint/stylelint/pull/4797#issuecomment-637646864), so we have a better understanding of what's going on now.

There's a workaround fix pending release, so I'll close this issue for now.

> The demo is working but probably because the playground is not using the last Stylelint version?

The demo is updated as part of the release process, and so should be using the latest version of stylelint.
